File: C#/Problems/Easy/Problem1389/Solution.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LeetCode.Problems.Easy.Problem1389
{
    public class Solution {
        public int[] CreateTargetArray(int[] nums, int[] index) {
            if (nums.Length == 0 || index.Length == 0) {
                return new int[]{};
            }
            
            var target = new List<int>();
            
            for (int i = 0; i < index.Length; i++) {
                var idx = index[i];
                var value = nums[i];
                
                target.Insert(idx, value);
            }
            
            return target.ToArray();
        }
    }
}

File: Golang/problems/easy/problem1389/solution.go
package solutions

// CreateTargetArray function is a solution for the
// Problem #1389. Create Target Array in the Given Order

// from leetcode.com
func CreateTargetArray(nums []int, index []int) []int {
	var target []int

	for i := 0; i < len(index); i++ {
		var idx, value = index[i], nums[i]

		if idx == 0 {
			target = append([]int{value}, target...)
		} else if len(target) == idx {
			target = append(target, value)
		} else if len(target) > 0 && idx < len(target) {
			target = append(target[:idx+1], target[idx:]...)
			target[idx] = value
		}
	}

	return target
}
